cache/cache.go

```diff
@@ -243,8 +240,26 @@ func readWatchChannel(
 			}
 			return err
 		}
+
```

Comment:
This code should be in demux.Broadcast not here.

Reply:
One option is to rename the current `Broadcast` to `broadcast` and have a new `Broadcast` function that batches input.